engine/src/main/java/org/terasology/rendering/dag/stateChanges/EnableStateParameter.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.stateChanges;
+
+import static org.lwjgl.opengl.GL11.glEnable;
+
+/**
+ * TODO: Add javadocs
+ */
+abstract class EnableStateParameter extends SetStateParameter {
```

Comment:
Because of the process() method.

The EnableStateParameter calls glEnable(), and DisableStateParameter calls glDisable(). The best I can do is remove SetStateParameter (I wanted to do it either way..) to simplify things a little.